Python/Python.sublime-syntax

```diff
@@ -402,6 +402,120 @@ contexts:
+        3: punctuation.separator.annotation.type-comment.python
+      embed: type-comment
+      escape: '\n'
+      embed_scope: comment.line.type-hint.python
+
+    - include: comments_noannotation
+
+  comments_afterignore:
+    - meta_scope: comment.line.type-hint.python
+    - include: comments_noannotation
+    - match: \n
+      pop: true
+    - match: \[
+      # Mypy's # type: ignore[error-code1, error-code2]
+      scope: punctuation.section.brackets.begin.python
+      set:
+        - meta_scope: comment.line.type-hint.python
+        - match: \[
+          scope: invalid.illegal.python
+        - match: '[a-zA-Z_-][a-zA-Z_0-9-]+'
+          scope: comment.line.type-hint.python.mypy-errorcode
+        - match: '[0-9]'
+          scope: invalid.illegal.python
+        - match: ','
+          scope: punctuation.separator.sequence.python.mypy-errorcode
+        - match: \]
+          scope: punctuation.section.brackets.end.python
+          set:
+              # Don't allow a second block.
+            - match: \[
+              scope: invalid.illegal.python
+              pop: true
+            - include: comments_afterignore
+    - match: \S
+      scope: invalid.illegal.python
+
+  type-comment:
+    - include: comments_noannotation
+
+    # Don't use expressions here, only a few types are reasonable
+    # as annotations.
+    - include: constants
+    - include: builtin-exceptions
+    - include: builtin-types
+    - include: builtin-functions
+    - match: '{{identifier}}'
+      scope: support.class.python
+
+    - match: '\s*(\[)'
+      captures:
+        1: meta.item-access.python punctuation.section.brackets.begin.python
+      push:
+        - meta_content_scope: meta.item-access.arguments.python
+        - match: \]
+          scope: meta.item-access.python punctuation.section.brackets.end.python
+          pop: true
+        - match: ':'
+          scope: punctuation.separator.slice.python
+        - include: type-comment
+
+    - match: '\('
+      scope: punctuation.section.parens.begin.python
+      push:
+        - meta_content_scope: meta.function.parameters.python
+        - match: \)
+          scope: punctuation.section.parens.end.python
+          pop: true
+        - match: \(
+          scope: invalid.illegal.stray.brace.round.python
+
+        - match: \*{3,}
+          scope: invalid.illegal.syntax.python
+        - match: \*\*
+          scope: keyword.operator.unpacking.mapping.python
+        - match: \*
+          scope: keyword.operator.unpacking.sequence.python
+
+        - match: '->'
+          scope: invalid.illegal.stray.return.arrow.python
+        - include: type-comment
+
+    - match: '->'
+      scope: punctuation.separator.sequence.python meta.function.return-type.python
+      set:
+        - match: '->'
+          scope: invalid.illegal.stray.return.arrow.python
+        - include: type-comment
+        - meta_scope: meta.function.return-type.python
```

Comment:
This should be moved up 3 lines so it is the first entry in the anonymous context